test/sharness/t0040-add-and-cat.sh

```diff
@@ -505,7 +505,7 @@ test_add_cat_expensive "--cid-version=1" "zdj7WcatQrtuE4WMkS4XsfsMixuQN2po4irkYh
 # encoded with the blake2b-256 hash funtion
 test_add_cat_expensive '--hash=blake2b-256' "zDMZof1kwndounDzQCANUHjiE3zt1mPEgx7RE3JTHoZrRRa79xcv"
 
-test_add_named_pipe " Post http://$API_ADDR/api/v0/add?encoding=json&progress=true&r=true&stream-channels=true:"
```

Comment:
This worries me. Will this break anything?

Reply:
@whyrusleeping thoughts?

Reply:
I checked back with why before changing this. The point is simply that when go-ipfs-cmds/cli fills in the default values in the request, and this request is passed on to the client in go-ipfs-cmds/http, which then makes a request that contains the default value for unset options. That's why they show up in the error message. Making requests without explicitly stating options with defaults still works, but we just don't happen to do it this way. This does not break compat to other clients.

Reply:
but yeah we might add a test for that here, sorry

Reply:
Np. When I saw this change, I thought we might be changing the flags. However, this is totally reasonable.